blockprod/src/detail/mod.rs

```diff
@@ -435,36 +458,27 @@ impl BlockProduction {
                 ));
             }
 
-            // TODO: see if we can simplify this
-            let transactions = match transactions_source.clone() {
-                TransactionsSource::Mempool => {
-                    // We conservatively use the minimum timestamp here in order to figure out
-                    // which transactions are valid for the block.
-                    // TODO: Alternatively, we can construct the transaction sequence from the
-                    // scratch every time a different timestamp is attempted. That is more costly
-                    // in terms of computational resources but will allow the node to include more
-                    // transactions since the passing time may release some time locks.
-                    let accumulator = self
-                        .collect_transactions(
-                            current_tip_index.block_id(),
-                            min_constructed_block_timestamp,
-                        )
-                        .await?;
-                    match accumulator {
-                        Some(acc) => acc.transactions().clone(),
-                        None => {
-                            // If the mempool rejects the accumulator (due to tip mismatch, or otherwise), we should start over and try again
-                            log::info!(
-                                "Mempool rejected the transaction accumulator. Restarting the block production attempt."
-                            );
-                            continue;
-                        }
-                    }
-                }
-                TransactionsSource::Provided(txs) => txs,
+            // We conservatively use the minimum timestamp here in order to figure out
+            // which transactions are valid for the block.
+            // TODO: Alternatively, we can construct the transaction sequence from
+            // scratch every time a different timestamp is attempted. That is more costly
+            // in terms of computational resources but will allow the node to include more
+            // transactions since the passing time may release some time locks.
```

Comment:
Going through the above code and then this change, I'm thinking that the minimum here should be `MIN(last_timestamp_seconds_used, min_constructed_block_timestamp)` given that transactions may be earlier than `min_constructed_block_timestamp`.

`min_contstructed_block_timestamp` should probably by locally scoped to the construction of  `max_constructed_block_timestamp` as it's only used to calculate that value.